cluster/points_writer.go

```diff
@@ -280,6 +282,7 @@ func (w *PointsWriter) writeToShard(shard *meta.ShardInfo, database, retentionPo
 			// If the write returned an error, continue to the next response
 			if err != nil {
 				w.Logger.Printf("write failed for shard %d on node %d: %v", shard.ID, nodeID, err)
+				writeError = err
```

Comment:
This looks like last-error-wins, with regards to what goes up the layers. Is this what you want?